### ISSUE TITLE
Prepare extensions for Bionic Beaver updates

### ISSUE THREAD

**Issue by BronsonQuick** [Chassis/Chassis#657]
The newer version of Puppet doesn't allow classes to have [hyphens](https://coderwall.com/p/fvqsfw/avoid-hyphens-in-puppet-class-names) as it treats them as `minus` when it parses the manifests.

To get around this we should add in a `version: 2.1` to Chassis core then add in some logic to delete old extensions that contain a hyphen and upgrade them to underscores.

I think if we add that version and take code from here: https://github.com/Chassis/Chassis/pull/532/files#diff-34c580b66e353a044fcf7eee62a6b191R179-R195 we should be able to handle that. Once we have a PR for that merged, we can update the extensions that currently contain a hyphen with underscores and we'll be prepared for the Bionic Beaver update which we'll call version 3.

Old yaml files that have say `query-monitor` etc will get a redirect to `query_monitor` from Github so we should be fully backwards compatible.

**Comment by roborourke** [comment]
Cheers for working on this. What's the overall requirement here?

- do repos with hyphens need to be renamed on GH?
- If the extension still references v2 will it fail gracefully?

**Comment by BronsonQuick** [comment]
>do repos with hyphens need to be renamed on GH?
Correct Sir. I've added issues in all the repos for that.
>If the extension still references v2 will it fail gracefully?
Yep!

I outlined those details in the PR: https://github.com/Chassis/Chassis/pull/658

**Comment by roborourke** [comment]
Cheers @BronsonQuick. I wanna make sure we can roll this out with minimal disruption, tricky/impossible as the extensions aren't tagged!

At least if there's notifications of which extensions need updating in the output should enable folks to sort it out.

**Comment by BronsonQuick** [comment]
Yeah I'm keen to make this seamless as well.

So at the moment here's what it'll do when this is merged and we ship version 3.0 of Chassis:

1. It'll check the yaml files for any extensions that contain a hyphen.
2. It'll delete those folders and then recursively clone the extensions by swapping the hyphen for an underscore on the fly.
3. If the user has run `vagrant provision` as the command to trigger this then Puppet will check the state of the machine and adjust the state as required.

If the user has run any other `vagrant *` commands only steps 1 and 2 will occur.

I didn't want it to do notifications and rely on the user updating the extensions so I've automated it. If you're against this then I'm cool for feedback from you and @rmccue too and I can rework it when we've got a consensus. Cheers for looking at this. I'll be doing the PR's for the extensions later in the week so they are ready to roll. The Github automatic redirects when we rename the repos means backwards compat of cloning is all good.

Thanks for reviewing this @roborourke! I'm keen to get Bionic Beaver shipped in the next couple of weeks as I'm probably going to be off the grid for a couple of months and I wanna make sure Chassis moves forward onto Bionic Beaver.
